test/transport/PayloadTest.cpp

```diff
@@ -276,6 +289,33 @@ SCENARIO("test empty payload serialization after some data") {
         REQUIRE(dst2.copy() == dst2);
     }
 }
+
```

Comment:
can you please add a test for `shareRegion()` and `shareAll()`?

Reply:
added